### PR TITLE
Add a MA demo solving Helmholtz PDE

### PR DESCRIPTION
**Pull request by stephankramer**
As previously discussed now with Hessian-based monitor

### CODE REVIEW

File: demos/monge_ampere_helmholtz.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+# As in the `previous Monge-Ampère demo <./monge_ampere1.py.html>`__, we use the
+# MongeAmpereMover to perform the mesh movement based on this monitor. We need
+# to provide the monitor as a callback function that takes the mesh as its
+# input just as we have defined it above. During the iterations of the mesh
+# movement process the monitor will then be re-evaluated in the (iteratively)
+# moved mesh nodes so that, as we improve the mesh, we can also more accurately
+# express the monitor function in the desired high-resolution areas.
+
+mover = MongeAmpereMover(mesh, monitor, method="quasi_newton")
+mover.move()
```
Comment (jwallwork23) [inline demos/monge_ampere_helmholtz.py:147]: What do you think about including what's printed to screen in a separate code block? i.e. showing the convergence of the different metrics. We do this in Thetis demos.

File: demos/monge_ampere_helmholtz.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+# .. figure:: monge_ampere_helmholtz-adapted_mesh.jpg
+#    :figwidth: 60%
+#    :align: center
+#
+# Now let us see whether the numerical error has actually been reduced
+# if we solve the PDE on this mesh:
+
+u_h = solve_helmholtz(mover.mesh)
+error = u_h - u_exact(mover.mesh)
+print("L2-norm error on moved mesh:", sqrt(assemble(dot(error, error) * dx)))
```
Comment (jwallwork23) [inline demos/monge_ampere_helmholtz.py:167]: Again, I think it'd be nice for the printed error value to actually appear in the html.

**Review by jwallwork23 — CHANGES_REQUESTED** [review]
Thanks very much for this @stephankramer!

A few fairly minor comments. Also, please could you put a link to this demo at the end of the previous one, with a short sentence? So that it's easy to read through the demos on the webpage.

**Comment by jwallwork23** [comment]
I guess for the tests to pass we need to use the same Docker env as for Animate/Goalie. All the more reason to get https://github.com/pyroteus/animate/pull/47 working!

**Comment by jwallwork23** [comment]
> I guess for the tests to pass we need to use the same Docker env as for Animate/Goalie. All the more reason to get [pyroteus/animate#47](https://github.com/pyroteus/animate/pull/47) working!

Would you mind switching the Docker image used by Movement as part of this PR? If you switch it to the same one as we use for Animate and Goalie then the tests should pass.

File: demos/monge_ampere_helmholtz.py (hunk 1 above)
Comment (stephankramer) [reply demos/monge_ampere_helmholtz.py:147]: good idea. Out of interest why do we refer to the standard deviation of cell areas as "equidistribution" ?

Comment (jwallwork23) [reply demos/monge_ampere_helmholtz.py:153]: Hm good point, not sure how that ended up in there. Standard deviation would be a better term. We can either fix it here or open an issue for it, whichever you prefer.

**Review by jwallwork23 — APPROVED** [review]
Looks great, thanks again for this!

File: demos/monge_ampere_helmholtz.py (hunk 1 above)
Comment (stephankramer) [reply demos/monge_ampere_helmholtz.py:153]: Any objection to UTF -8 in terminal output? "Standard Deviation" is a bit long, and not entirely accurate - all other terminology (https://en.wikipedia.org/wiki/Coefficient_of_variation) is a bit obscure so I thought to clarify with σ/μ

Comment (jwallwork23) [reply demos/monge_ampere_helmholtz.py:153]: Fine by me. Nice! Even better.

**Pull request merged**